## Le stress améliore t-il ou dégrade t-il les performances scolaires ? 

## ⚙️ Régression linéaire, Corrélation et Test (Python)

### ➡️Importation et aperçu de notre dataset

In [1]:
import pandas as pd

data = pd.read_csv("Data/StressLevelDataset_S2.csv")
data.head()

,anxiety_level,self_esteem,mental_health_history,depression,headache,blood_pressure,sleep_quality,breathing_problem,noise_level,living_conditions,...,basic_needs,academic_performance,study_load,teacher_student_relationship,future_career_concerns,social_support,peer_pressure,extracurricular_activities,bullying,stress_level
0,14,20,0,11,2,1,2,4,2,3,...,2,3,2,3,3,2,3,3,2,1
1,15,8,1,15,5,3,1,4,3,1,...,2,1,4,1,5,1,4,5,5,2
2,12,18,1,14,2,1,2,2,2,2,...,2,2,3,3,2,2,3,2,2,1
3,16,12,1,15,4,3,1,3,4,2,...,2,2,4,1,4,1,4,4,5,2
4,16,28,0,7,2,3,5,1,3,2,...,3,4,3,1,2,1,5,0,5,1


### ➡️ Valeurs manquantes et nettoyage

In [2]:
print(data.info())
print(data.describe())

<class 'pandas.DataFrame'>
RangeIndex: 1100 entries, 0 to 1099
Data columns (total 21 columns):
 #   Column                        Non-Null Count  Dtype
---  ------                        --------------  -----
 0   anxiety_level                 1100 non-null   int64
 1   self_esteem                   1100 non-null   int64
 2   mental_health_history         1100 non-null   int64
 3   depression                    1100 non-null   int64
 4   headache                      1100 non-null   int64
 5   blood_pressure                1100 non-null   int64
 6   sleep_quality                 1100 non-null   int64
 7   breathing_problem             1100 non-null   int64
 8   noise_level                   1100 non-null   int64
 9   living_conditions             1100 non-null   int64
 10  safety                        1100 non-null   int64
 11  basic_needs                   1100 non-null   int64
 12  academic_performance          1100 non-null   int64
 13  study_load                    1100 non-null 

In [5]:
print(data.isna().sum())

anxiety_level                   0
self_esteem                     0
mental_health_history           0
depression                      0
headache                        0
blood_pressure                  0
sleep_quality                   0
breathing_problem               0
noise_level                     0
living_conditions               0
safety                          0
basic_needs                     0
academic_performance            0
study_load                      0
teacher_student_relationship    0
future_career_concerns          0
social_support                  0
peer_pressure                   0
extracurricular_activities      0
bullying                        0
stress_level                    0
dtype: int64


### ➡️ Corrélation ( Pearson )

In [16]:
r, p = pearsonr(data["stress_level"], data["academic_performance"])
print(f"Corrélation r = {r:.3f}, p-value = {p:.3f}")

Corrélation r = -0.721, p-value = 0.000


### ➡️Modèle de régression de linéaire

In [19]:
X = sm.add_constant(data["stress_level"])
model = sm.OLS(data["academic_performance"], X).fit()
print(model.summary())

                             OLS Regression Results                             
Dep. Variable:     academic_performance   R-squared:                       0.520
Model:                              OLS   Adj. R-squared:                  0.519
Method:                   Least Squares   F-statistic:                     1188.
Date:                  Mon, 29 Jun 2026   Prob (F-statistic):          4.56e-177
Time:                          20:06:28   Log-Likelihood:                -1538.5
No. Observations:                  1100   AIC:                             3081.
Df Residuals:                      1098   BIC:                             3091.
Df Model:                             1                                         
Covariance Type:              nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            4.0094     

### ➡️ Résumé

Les données montrent une relation négative forte entre le stress et la performance scolaire (r = –0.721, p < 0.001).
Le modèle de régression confirme que le stress réduit significativement la performance : chaque unité de stress diminue le score académique d’environ 1.24 points (β = –1.24).
Avec un R² de 0.520, le stress explique plus de la moitié de la variance des performances, ce qui en fait un facteur majeur.

Une nuance importante : dans la littérature, un stress léger peut parfois agir comme un stimulant — il favorise l’attention, l’énergie, et la mobilisation cognitive.

Mais dans ce dataset, cette zone “optimale” n’apparaît pas : la relation est globalement et fortement négative, ce qui suggère que les niveaux de stress observés dépassent probablement le seuil où il devient bénéfique.

Conclusion : Le stress n’est pas seulement un inconfort : c’est un frein mesurable à la performance scolaire. Même si un stress léger peut théoriquement être stimulant, les données montrent que dans la majorité des cas, il dégrade la performance de manière significative.